### DISCUSSION TITLE
Connection pooling and long-running requests

### DISCUSSION THREAD

**Discussion by mkjpryor** [encode/httpx#2815] [Q&A]
I have an application that does a lot of Kubernetes watches, which means a lot of long-lived HTTP connections. This results in a lot of PoolTimeout errors :-)

What is the recommended thing to do here? Massively increase the size of the connection pool? Have a client per connection?

**Comment by lovelydinosaur** [comment]
> Have a client per connection?

No. Having a client-per-connection won't gain you anything.

> increase the size of the connection pool?

If you've got a specific use case where you want lots of connections but they're mostly idle, then that seems sensible. (?)

**Comment by mkjpryor** [comment]
The problem I have is that the connections consumed by these long-lived requests stay ACTIVE (this is a form of long-polling, essentially). So the total number of objects that the thing can possibly handle is limited by the size of the connection pool. This thing needs to be able to handle potentially 1000s objects.

**Reply by karpetrosyan** [reply]
You can try increasing the size of the connection pool.
